internal/controller/runtime/runtime.go

```diff
@@ -0,0 +1,99 @@
+package runtime
+
+/*
+Copyright 2020 Crunchy Data
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"os"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/crunchydata/postgres-operator/internal/controller/postgrescluster"
+	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1alpha1"
+)
+
+const (
+	// the name of the event recorder created for the postgrescluster controller
+	recorderName = "postgrescluster-controller"
+
+	// default refresh interval in minutes
+	refreshInterval = 60
+)
+
+// CreateRuntimeManager creates a new controller runtime manager for the PostgreSQL Operator.  The
+// manager returned is configured specifically for the PostgreSQL Operator, and includes any
+// controllers that it will be responsible for managing PostgreSQL clusters using the
+// 'postgrescluster' custom resource.
+func CreateRuntimeManager() (manager.Manager, error) {
+
+	pgoScheme, err := createPGOScheme()
+	if err != nil {
+		return nil, err
+	}
+
+	// create controller runtime manager
+	syncPeriod := refreshInterval * time.Minute
+	mgr, err := manager.New(config.GetConfigOrDie(), manager.Options{
+		Namespace:  os.Getenv("PGO_TARGET_NAMESPACE"), // if empty then watching all namespaces
```

Comment:
`GetConfigOrDie()` and `os.Getenv()` seem like responsibilities that belong in `main()`. This file is good to go, but I'm going to think on it some more.

Reply:
`os.Getenv()` has been moved to `main()`, and `GetConfig()` is now used instead to capture the error and propagate to `main()`.